File: C/0342-power-of-four.c
bool isPowerOfFour(int n) {
    if(n == 0)
        return false;

    if(n == 1)
        return true;

    long mask = 4;

    while(mask < INT_MAX) {
        if(n == mask)
            return true;

        mask = mask << 2;
    }

    return false;
}

File: dev-rust/0342-power-of-four.rs
impl Solution {
    pub fn is_power_of_four(n: i32) -> bool {
        if n == 0 {
            return false;
        } 

        if n == 1 {
            return true;
        }

        let mut mask: i64 = 4;
        let n_64: i64 = n.into();

        while mask < (i32::MAX as i64) {
            if n_64 == mask {
                return true;
            }

            mask <<= 2;
        }

        return false;

    }
}